driver-core/src/main/com/mongodb/internal/connection/ClientMetadataHelper.java

```diff
@@ -102,69 +80,95 @@ private static boolean nameMatches(final String name, final String... prefixes)
         return false;
     }
 
-    @Nullable
-    static BsonDocument createClientMetadataDocument(@Nullable final String applicationName) {
-        return createClientMetadataDocument(applicationName, null);
-    }
-
-    @Nullable
-    public static BsonDocument createClientMetadataDocument(@Nullable final String applicationName,
+    public static BsonDocument getClientMetadataDocument(@Nullable final String applicationName,
                                                             @Nullable final MongoDriverInformation mongoDriverInformation) {
-        return createClientMetadataDocument(applicationName, mongoDriverInformation, CLIENT_METADATA_DOCUMENT);
-    }
+        if (cachedClientMetadataDocument != null) {
+            // "asDocument" prevents:
+            // MS_EXPOSE_REP: Public static method may expose internal representation by returning array
+            return cachedClientMetadataDocument.asDocument();
+        }
 
-    @Nullable
-    static BsonDocument createClientMetadataDocument(@Nullable final String applicationName,
```

Comment:
It is not possible for the client document to be null because at least the os fields will be populated with "unknown"